java/AccelerometerParser.java

```diff
@@ -187,7 +180,23 @@ public static void main(String[] args) {
 				} else {
 					System.err.println("unknown parameter " + funcName + ":" + funcParam);
 				}
+			}
 
+			if (!startTimeStr.isEmpty()) {
+				try {
+					startTime = dateFormat.parse(startTimeStr).getTime();
+				} catch (ParseException ex) {
+					System.err.println("error parsing startTime:'" + startTimeStr + "', must be in format: 1996-7-30T13:59");
+					System.exit(-2);
+				}
+			}
+			if (!endTimeStr.isEmpty()) {
+				try {
+					endTime = dateFormat.parse(endTimeStr).getTime();
+				} catch (ParseException ex) {
+					System.err.println("error parsing endTime:'" + endTimeStr + "', must be in format: 1996-7-30T13:59");
+					System.exit(-2);
+				}
```

Comment:
Any particular reason why this is moved to the bottom?

Reply:
I know. The main reason for this is that we can't guarantee that lateTime or startTime flag will be read after the timeZone flag. 

That's why we should only do the conversion after all the flags have been read.